### DISCUSSION TITLE
directly open a saved view

### DISCUSSION THREAD

**Discussion by Paolo-lab** [iTwin/itwinjs-core#5983] [Q&A] [answered]
Hello everyone good morning

my client asks me if it is possible to start the application instead of from the model selected within the project, from a saved view
When opening the viewer I managed to switch to the saved view but anyway there is a refresh as it opens the model first and then the view immediately.

Thanks in advance

**Comment by aruniverse** [comment]
This should be possible.

If you know which saved view you want to open up before going to the viewer, you can modify the `viewCreatorOptions` passed into the `<Viewer />` component.

eg:

```tsx
 const viewCreatorOptions = useMemo(
    () => ({
      viewportConfigurer: async (vp: ScreenViewport) => {
        if (savedView) {
          const viewStateProps = savedView.savedViewData.legacyView;
          const viewState = SpatialViewState.createFromProps(viewStateProps, vp.iModel);
          await viewState.load();
          vp.changeView(viewState);
        }
      },
    }),
    [savedView]
  );

...

    <Viewer
	  ...
      viewCreatorOptions={viewCreatorOptions}
    />
```

Note, here im using the legacy saved view data, but should work with the new APIs as well
https://developer.bentley.com/apis/savedviews/operations/get-savedview/#savedviewwithdata

**Comment by Paolo-lab** [comment]
hi good morning thanks for the reply
I'm trying to use your code but I'm having trouble with "viewStateProps"
I imagined taking the view saved through the getByIdAsync API with the view id inside.
then using the "itwin3dView" method which returns a "SavedViewItwin3dViewInterface" object which I cannot transform into "ViewStateProps", I'm sure I missed some steps

thanks Paul
<img width="693" alt="saveview" src="https://github.com/iTwin/itwinjs-core/assets/57446652/a5fc5ec0-1607-489b-b37b-67a18547965d">

**Reply by hnn0003** [reply]
Hey!
If the saved view you're grabbing is a legacy view (ie a saved view not created using the new APIs), then it should be relatively straight forward:
```
if (savedView.savedViewData.legacyView !== undefined) {
    const viewStateProps: ViewStateProps = {
      viewDefinitionProps: savedView.savedViewData.legacyView.viewDefinitionProps,
      categorySelectorProps: savedView.savedViewData.legacyView.categorySelectorProps,
      modelSelectorProps: savedView.savedViewData.legacyView.modelSelectorProps,
      displayStyleProps: savedView.savedViewData.legacyView.displayStyleProps,
    };
}
```
This should give you a viewStateProps value that you can then use the SpatialViewState.createFromProps function to convert into a viewState.

If you do not have the legacyView data in your savedViewData (which means that the saved view was created directly using our APIs instead of through Design Review, PlantSight, etc), then it gets more complicated. We've changed how we're storing the data a little bit for an 'API view' (ie, a view created using the APIs), so in that case, the itwin3dView should have all of the data you need, but it would have to be transformed into what a ViewStateProps is expecting. 

We should be able to help you do that transformation if you need to do so, but my guess would be that what you actually need is to just get the information through the savedView.savedViewData.legacyView

**Comment by Paolo-lab** [comment, marked as answer]
Hi, I was finally able to load the views saved through legacy. I had problems understanding why the legacy part with the various interfaces was not developed in the example you make available for us mere mortals :-) I understand it because it also says that it will become a bit obsolete. Once I understood that it was necessary to implement all the interfaces everything went well, although I must say that there is always a slight loading of the main view before loading the viewCreatorOptions

thanks thanks thanks
